metrics/metrics_test.go

```diff
@@ -0,0 +1,70 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var defaultRegistry = metrics.Registry
```

Comment:
I see that you call `RegisterMetrics()` twice in this file, both of which act against `metrics.Registry` (as far as I understand). Is this not a problem, and if so why not?

Reply:
Removed the calls in favor of cleaner implementation.

It wasn't a problem because `RegisterMetrics()` was acting against `metrics.Registry` only once. Consequent calls would just return realizing that it was already called before.